compose: Redesign mobile message buttons popover.

sayamsamal opened the pull request:
As part of the popover menu redesign, this redesigns the mobile message buttons popover using the new "popover-menu" tippy theme and improves accessibility by using appropriate ARIA attributes.

This PR also updates the logic for the mobile message buttons popover to mimic the "Start new conversation" button's logic — show new topic option only when in a stream narrow, else show the new channel message option. And removes the `is_in_private_narrow` check from the logic, as we show the new direct message option regardless of it.

Fixes part of #28699.

[CZO Discussion](https://chat.zulip.org/#narrow/stream/101-design/topic/compose.3A.20Redesign.20mobile.20message.20buttons.20popover.20.2330603/near/1839459)

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

| Desc | Before | After |
|--------|--------|--------|
| Light Mode | <img width="302" alt="mobile-msg-popover-light-before" src="https://github.com/zulip/zulip/assets/82862779/40cfc644-27e2-4691-b37f-4cd26f21c6fe"> | <img width="302" alt="mob-msgs-update-after-light" src="https://github.com/zulip/zulip/assets/82862779/a6106fb6-42d4-4343-b754-91fc6c21f304"> |
| Dark Mode | <img width="302" alt="mobile-msg-popover-dark-before" src="https://github.com/zulip/zulip/assets/82862779/02fe6080-7eeb-4ee2-b104-7c4bf92bfe6c"> | <img width="302" alt="mob-msgs-update-after-dark" src="https://github.com/zulip/zulip/assets/82862779/a326825e-4e66-474e-b315-0381fd280398"> |

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
lgtm aside from the `prettier` error. Please post in #design for feedback there as well.

sayamsamal commented:
Oh seems like I didn't have the git pre-commit hook setup. Fixed the error.

sayamsamal commented:
@timabbott this is ready for another review.

Updates:
- Changed from the conditional menu option "New channel message"/"New topic" -> a static "Start new conversation" abstracting the behavior from the UI.
- Added the hotkey hints.

alya commented:
We still need to remove the keyboard shortcut from this tooltip, since the button opens a menu.

![Screenshot 2024-06-28 at 09 14 59](https://github.com/zulip/zulip/assets/2090066/95d4d8bb-efad-49a9-a5e1-79bc9777ce53)

alya commented:
Testing it out, I'm happy with the "Start new conversation" label, especially since you could just as well use that menu option to start a DM or a channel thread.

timabbott commented:
Merged, thanks @sayamsamal!
